### ISSUE TITLE
Transpiling of /(\1)+\1\1/u to /(\x01)+\1\1/

### ISSUE THREAD

**Issue by nhahtdh** [mathiasbynens/regexpu#19]
Your library transpile `/(\1)+\1\1/u` to `/(\x01)+\1\1/`.

Is there a change in the ES6 specs which allows such interpretation?

---

The current behavior in ES5:
```
var string = '\x01\x01';
var match = string.match(/(\1)+\1\1/);
console.log(match);
-> [ "", "" ]
```
All of them are interpreted as capturing groups, as far as my testing reveals on Firefox. The ECMA 5 spec also seems to agree with this particular case: http://www.ecma-international.org/ecma-262/5.1/#sec-15.10.2.11

The specs seems to allow `\1` to appear before its capturing group as long as there are enough number of capturing groups in the entire expression.

> It is an error if n is greater than the total number of left capturing parentheses in the entire regular expression.

**Comment by mathiasbynens** [comment]
I believe this is a bug in regjsparser: https://github.com/jviereck/regjsparser/issues/70

**Comment by jviereck** [comment]
Thanks for reporting this! From out of my head I think the current implementation of the `regjsparser` checks the so far parsed count of groups to decide if `\1` is a back-reference or a octal match.

Here are a few results from playing around with capturing matches from inside of Firefox Console:

```js
// Simple reduction of the reported bug.
r = /(\1)a/.exec('\x01a'); // -> [ "a", "" ]
r[1].length == 0 // true

// A little bit more advanced.
r = /(\2)(b)a/.exec('\x02ba');
r.length == 0;

// Q: What happens if the back-reference is larger than total number of left capturing parentheses?
// A: Interpret the \2 as an octal match.
/(\2)a/.exec('\x01a'); // -> null
/(\2)a/.exec('\x02a'); // ['\x02a', '\x02']
```
Will discuss the idea to fix this bug in the jviereck/regjsparser#70 issue.

**Comment by mathiasbynens** [comment]
https://github.com/jviereck/regjsparser/issues/70 is now fixed.
